Lib/AssetCache.php

```diff
@@ -43,40 +43,93 @@ public function write($build, $content) {
  * @param string $target The target file being built.
  * @return boolean
  */
-	public function isFresh($target) {
-		$ext = $this->_Config->getExt($target);
-		$files = $this->_Config->files($target);
+    public function isFresh($target) {
+        $ext = $this->_Config->getExt($target);
+        $files = $this->_Config->files($target);
 
-		$theme = $this->_Config->theme();
-		$target = $this->buildFileName($target);
+        $theme = $this->_Config->theme();
+        $target = $this->buildFileName($target);
 
-		$buildFile = $this->_Config->cachePath($ext) . $target;
+        $buildFile = $this->_Config->cachePath($ext) . $target;
 
-		if (!file_exists($buildFile)) {
-			return false;
-		}
-		$configTime = $this->_Config->modifiedTime();
-		$buildTime = filemtime($buildFile);
+        if (!file_exists($buildFile)) {
+            return false;
+        }
+        $configTime = $this->_Config->modifiedTime();
+        $buildTime = filemtime($buildFile);
 
-		if ($configTime >= $buildTime) {
-			return false;
-		}
+        if ($configTime >= $buildTime) {
+            return false;
+        }
 
-		$Scanner = new AssetScanner($this->_Config->paths($ext, $target), $theme);
+        $Scanner = new AssetScanner($this->_Config->paths($ext, $target), $theme);
 
-		foreach ($files as $file) {
-			$path = $Scanner->find($file);
-			if ($Scanner->isRemote($path)) {
-				$time = $this->getRemoteFileLastModified($path);
-			} else {
-				$time = filemtime($path);
-			}
-			if ($time === false || $time >= $buildTime) {
-				return false;
-			}
-		}
-		return true;
-	}
+        foreach ($files as $file) {
+            $path = $Scanner->find($file);
+            if ($Scanner->isRemote($path)) {
+                $time = $this->getRemoteFileLastModified($path);
+            } else {
+                $time = filemtime($path);
+            }
+            if ($time === false || $time >= $buildTime) {
+                return false;
+            }
+
+            // If this is a SASS or LESS file, check any imports defined to see if they are not fresh.
```

Comment:
@sprak3000 At first I wondered if we could reliably count on the extension vs. checking what the engine is, but it looks like the extensions are strict for these frameworks: 

e.g. https://teamtreehouse.com/community/does-every-sass-file-really-need-to-end-in-scss